### PR TITLE
Introduce workaround to show something resembling a back button in iOS

### PR DESCRIPTION
**Pull request by mokagio**
See discussion in https://github.com/woocommerce/WooCommerce-Shared/issues/61

The problem seems to be that the JS bundle generated by `make bundle-ios` does not include, or knows how to refer to, the assets that the react-navigation library uses to render the back button in iOS (not sure why they don't access the system provided button, but there must be a reason).

I tried to change the wiring so that the assets became available, but got nowhere.

This workaround shows a chevron-like character in the navigation label to make up for the lack of button image.

image

Please note that I'm not arguing for this as a long term solution. But I have a feeling the issue might be deeper than expected and I thought this could be enough to get us over the line to validate the actual feature and React Native approach in the real world.

## To test

- Checkout this branch
- Run `DEV=1 make bundle-ios`
- Run `make xcode`
- Run the demo app from Xcode

### CODE REVIEW

File: ShippingZones.tsx [hunk 1]
```diff
@@ -100,13 +100,28 @@ const ShippingZonesList = () => {
...
+              ? <Text>{''}</Text>
+              // By returning undefined in non-iOS systems, we let the library do its thing.
+              : undefined
+          }}
+          label={
+            Platform.OS === 'ios'
+              ? `\u2329 ${backButtonLabel}`
+              : backButtonLabel
+          }
+          labelVisible={Platform.OS === 'ios' ? true : false}
```
Comment (mokagio) [inline ShippingZones.tsx:124]: If we don't want to have "Settings" for now (or ever), we can easily remove it, like so:

```suggestion
          label={Platform.OS === 'ios' ? `\u2329` : undefined}
          labelVisible={Platform.OS === 'ios' ? true : false}
```

image

**Comment by wzieba** [comment]
We can do this or we can consider using Stack Navigation (different than Native Stack Navigation we use right now) which might resolve the issue as well. I'll continue working on it in #56 and then we can compare which solution feels better (more native?). WDYT?

**Comment by mokagio** [comment]
@wzieba any solution that doesn't involved a dirty workaround such as this one is welcome 😄

**Comment by wzieba** [comment]
This issue has been addressed in #70 and back-navigation has been introduced in #54 . I'll allow myself to close this PR @mokagio . But thank you very much for your contribution and help with this issue 🙏 . This would be most likely our solution if @Ecarrion wouldn't fix the issue in #70.

**Pull request closed without merging**